src/providers/provider.ts

```diff
@@ -6,18 +6,36 @@ export abstract class Provider {
   static readonly placeholder: string = '#{}'
 
   matchWith(input: string): Completion[] {
-    // TODO: Improve filtering with weights
-    const suggestions = this.completions.filter(val => val.includes(input))
-      .map(sugg => {
-        return {category: this.category, value: sugg}
+    const inputLowered = input.toLowerCase()
+    const inputHasUpperCase = /[A-Z]/.test(input)
+
+    // TODO: Show lowercase first and the upper case, if search is case insensitive
+
+    const suggestions = this.completions
+      .filter((suggestion) => {
```

Comment:
It’s a matter of taste, it can be simplified as follows.

```ts
      .filter((suggestion) => inputHasUpperCase ? suggestion.includes(input) : suggestion.toLowerCase().includes(inputLowered)
```

Reply:
Indeed, it is more concise the ternary operator! Thanks for the suggestion

Reply:
I've discovered a function `localeCompare` which seems to sort correctly strings (starting with lowercase), so I've removed the part `.sort( (a, b) => Number(b.toLowerCase().startsWith(inputLowered)) - Number(a.toLowerCase().startsWith(inputLowered)))`. Let me know if the behaviour is same or better for you, if not I'll revert these changes.

Reply:
I didn't know `localCompare`, and it looks good!